### ISSUE TITLE
torch.library.opcheck doesn't check strides for CPU Tensors

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#149468]
Repro:
```py
import torch
from torchvision.transforms.functional import to_pil_image, pil_to_tensor
import PIL

def crop(pic, box):
    img = to_pil_image(pic.cpu())
    cropped_img = img.crop(box)
    return pil_to_tensor(cropped_img).to(pic.device) / 255.

img = torch.ones(3, 64, 64)
img *= torch.linspace(0, 1, steps=64) * torch.linspace(0, 1, steps=64).unsqueeze(-1)

cropped_img = crop(img, (10, 10, 50, 50))

def f(img):
    return crop(img, (10, 10, 50, 50))

cropped_img = f(img)
print(img.shape, img.stride())
print(cropped_img.shape, cropped_img.stride())

from typing import Sequence

@torch.library.custom_op("mylib::crop", mutates_args=())
def crop(pic: torch.Tensor, box: Sequence[int]) -> torch.Tensor:
    img = to_pil_image(pic.cpu())
    cropped_img = img.crop(box)
    result = (pil_to_tensor(cropped_img) / 255.).to(pic.device, pic.dtype)
    return result

@crop.register_fake
def _(pic, box):
    channels = pic.shape[0]
    x0, y0, x1, y1 = box
    # result = pic.new_empty(y1 - y0, x1 - x0, channels).permute(2, 0, 1)
    result = pic.new_empty(channels, y1 - y0, x1 - x0)
    return result

result = torch.library.opcheck(crop, (img, (10, 10, 50, 50)))
print(result)
```

cc @ezyang @gchanan @kadeng @msaroufim @janeyx99 @anjali411 @bdhirsh @penguinwu @bobrenjc93 @aorenste @malfet @xmfan @chauhang

**Comment by zou3519** [comment]
> [@zou3519](https://github.com/zou3519) would you be so kind, to please add labels to the issues you file? (It's really hard to make a judgement by just `high pri` label on whether or not problem is really a high pri)

Yes, will do
